script/check_coveragerc.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python3
+"""Remove lines from .coveragerc if they are at 100% coverage.
+
+Runs the full test suite and works out if any lines in .coveragerc
+don't need to be there.
+"""
+
+import fileinput
+import json
+import os
+
+import pytest
+
+COVERAGERC = ".coveragerc"
+TEMP_JSON = "build/_temp_coverage.json"
+
+
+def replace_all(file: str, before: str, after: str):
+    """Replace all instances of string in file."""
+    for line in fileinput.input(file, inplace=1):
+        print(line.replace(before, after), end="")
+
+
+def main():
+    """Run the script."""
+
+    if os.path.exists(TEMP_JSON):
+        os.remove(TEMP_JSON)
+
+    try:
+        # Turn off all .coveragerc ignoring
+        replace_all(COVERAGERC, "omit =", "omit_disabled =")
+
+        # Run tests in parallel for speed
+        pytest.main(
+            [
+                "-n",
+                "auto",
+                "tests/components/",
+                "--cov=homeassistant.components",
+                "--cov-report",
+                f"json:{TEMP_JSON}",
+            ]
+        )
+
+        # parse the json results file
+        with open(TEMP_JSON, encoding="UTF-8") as infile:
+            data = json.load(infile)
+
+        # make a list of 100% passing files
+        out_list = []
+        for filename, file_data in data["files"].items():
+            percent = file_data["summary"]["percent_covered"]
+            if percent == 100.0:
```

Comment:
When I did a similar thing in the past, we used a lower threshold.
I think we used 90% or maybe even 80%

The idea behind that was that if it was already at say 80%, then we should leverage coverage to warn on coverage reductions.

Reply:
Actually it was 97M% in https://github.com/home-assistant/core/pull/86466
See also https://github.com/home-assistant/core/pull/60349 and https://github.com/home-assistant/core/pull/60349